cli/lib/kontena/cli/nodes/update_command.rb

```diff
@@ -3,16 +3,35 @@ class UpdateCommand < Kontena::Command
     include Kontena::Cli::Common
     include Kontena::Cli::GridOptions
 
+    requires_current_master
+    requires_current_master_token
+    requires_current_grid
+
     parameter "NODE", "Node name"
+
     option ["-l", "--label"], "LABEL", "Node label", multivalued: true
+    option "--clear-labels", :flag, "Clear node labels"
+    option ["--token"], "TOKEN", "Node token"
+    option "--generate-token", :flag, "Generate new node token"
+
+    def update_token
+      data = {}
+
+      data[:token] = self.token if self.token
+
+      spinner "Updating node #{self.node.colorize(:cyan)} token" do
+        client.put("nodes/#{current_grid}/#{self.node}/token", data)
+      end
+    end
 
     def execute
-      require_api_url
-      require_current_grid
-      token = require_token
+      self.update_token if self.token or self.generate_token?
```

Comment:
Split out the `PUT /v1/nodes/:id/token` as `kontena node reset`, with appropriate confirm prompt:

```
> Resetting the node token will disconnect the agent (unless using --no-reset-connection), and require you to reconfigure the kontena-agent > Resetting the node token will disconnect the agent (unless using --no-reset-connection), and require you to reconfigure the kontena-agent > Resetting the node token will disconnect the agent (unless using --no-reset-connection), and require you to reconfigure the kontena-agent using the new `kontena node env` values before it will be able to reconnect. Are you sure? Yes
    *   Resetting node core-02 websocket connection token .. \DEBUG Excon opts: {:omit_default_port=>true, :connect_timeout=>5, :read_timeout=>30, :write_timeout=>5, :ssl_verify_peer=>true, :instrumentor=>Kontena::DebugInstrumentor}
DEBUG [Request]: PUT http://localhost:9292/v1/nodes/development/core-02/token | Headers: {Accept: application/json, Content-Type: application/json, Authorization: Bearer}  | Body: 
{"reset_connection":true}
    *   Resetting node core-02 websocket connection token .. |DEBUG [Response]: Headers: {Content-Type: application/json, X-Kontena-Version: 1.4.0.dev}  | Status: 200 | Body: 
{"id":"development/core-02","token":"lkQL...8g=="}
 [done] Resetting node core-02 websocket connection token     
```